src/Symfony/Component/Yaml/Parser.php

```diff
@@ -421,7 +421,7 @@ private function doParse($value, $flags)
                 }
 
                 // try to parse the value as a multi-line string as a last resort
-                if (0 === $this->currentLineNb) {
+                if (0 === $this->currentLineNb && 1 < $this->totalNumberOfLines && !$this->isNextLineIndented()) {
```

Comment:
```yaml
foo:
  bar
    baz
```

Will such a YAML string still work?

Reply:
the result should be `['foo' => 'bar baz']`

Reply:
indeed it doesn't. I'll add more tests and try to find another fix